[WIP] Automatic markdown-file-creator

ChristofJ95 opened the pull request:
This is the prototype of the automatic markdown file creator for the course Architecture of ML Systems. 
Simply execute it within the scripts/builtin directoy will iterate over it and represend feedback within the terminal. 
Furthermore, a markdown file for all valid headers will be created and placed in the same directory.

mboehm7 commented on scripts/builtin/mark_down_creator.py:
```diff
@@ -0,0 +1,385 @@
+import re
```
Missing license header - this is the reason why the build fails.

mboehm7 commented on scripts/builtin/mark_down_creator.py:
```diff
@@ -0,0 +1,385 @@
+# -----------------------------------------------------------------------------------------------------------------------------------------------------
+#   Name                            Type    Default     Meaning
+# -----------------------------------------------------------------------------------------------------------------------------------------------------
+#   SystemDs_Builtin_Markdown       .dm     -------     A .dm file of all files with a parsable header
```
you probably mean .md files - please fix the two typos (.dm).

mboehm7 commented on scripts/builtin/mark_down_creator.py:
```diff
@@ -0,0 +1,385 @@
+
+#class to make a pretty output possible
+class bcolors:
+    HEADER = '\033[95m'
```
On windows, unfortunately, only these color strings are output as is - it would be great to make this more robust or go back to basic output formatting.

mboehm7 commented on scripts/builtin/mark_down_creator.py:
```diff
@@ -0,0 +1,385 @@
+            new_file_data_array.append(entry)
+
+    for entry in incorrect_file_data:
+        print(f'{bcolors.WARNING}For the File: {bcolors.UNDERLINE}{bcolors.OKGREEN}' + entry + f'{bcolors.ENDC}{bcolors.WARNING} NO HEADER WAS FOUND AT ALL:{bcolors.ENDC}')
```
Please use consistent error messages - not all caps

mboehm7 replied:
Also it would be great to have an optional input argument to influence the error handling - if set to true, we could collect if there was at least one warning printed and if so throw an exception.

mboehm7 commented on scripts/builtin/mark_down_creator.py:
```diff
@@ -0,0 +1,385 @@
+    return new_file_data_array
+
+
+
```
remove unnecessary free lines (please consistently two between functions and otherwise one where needed)

mboehm7 reviewed (commented):
Thanks for the PR @ChristofJ95 - overall this is a great start and very helpful. Could you please address the minor cleanups, and then I would merge it in.

ChristofJ95 commented:
I did now implement all the needed fixes:

- changed '.dm' to '.md' in header
- removed unnecessary free lines
- use of consistent error message style
- colored output now working on windows and ubuntu (tested on my device)
- added global Flag to throw error in case one header does not follow the syntax rules

mboehm7 commented:
LGTM - thanks for the contribution @ChristofJ95. During the merge I only rename the python file and respective generated output.
